src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,122 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const InitialFormInfo: Movie = {
+    title: '',
+    description: '',
+    imdbUrl: '',
+    imdbId: '',
+    imgUrl: '',
+  };
+
+  const [formInfo, setFormInfo] = useState<Movie>(InitialFormInfo);
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = formInfo;
+
+  const validateUrl = (url: string): boolean => {
+    // eslint-disable-next-line
+    const regExUrl = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    return regExUrl.test(url.trim());
+  };
+
+  const validImgUrl = validateUrl(imgUrl);
+  const validImdbUrl = validateUrl(imdbUrl);
+
+  const handleInputChange = (value: string, field: string) => {
+    setFormInfo(prevInfo => ({ ...prevInfo, [field]: value }));
+  };
+
+  const clearForm = () => {
+    setFormInfo(InitialFormInfo);
+  };
+
+  const handleDataForm = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const data = { ...formInfo };
+
+    onAdd(data);
+    clearForm();
+    setCount(prevCount => prevCount + 1);
+  };
+
+  const checkIfDisabled = () => {
+    if (
+      !title.trim()
+      || !imdbId.trim()
+      || !imdbUrl.trim()
+      || !imgUrl.trim()
+    ) {
+      return true;
+    }
+
+    return false;
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleDataForm}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={handleInputChange}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={handleInputChange}
+
       />
 
       <TextField
         name="imgUrl"
         label="Image URL"
-        value=""
+        value={imgUrl}
+        onChange={handleInputChange}
+        regEx={validImgUrl}
```

Comment:
```suggestion
        isRegExValid={validImgUrl}
```
It's better to change the name accordingly so that it means what it has inside. As the name **regEx** means that it will contain some kind of a regEx, but you pass **boolean** insted